src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecurityUtils.cs

```diff
@@ -683,6 +779,54 @@ internal static string GetKeyDerivationAlgorithm(SecureConversationVersion versi
             }
         }
 
+        [SecuritySafeCritical]
+        internal static void FixNetworkCredential(ref NetworkCredential credential)
+        {
+            if (credential == null)
+            {
+                return;
+            }
+            string username = NetworkCredentialHelper.UnsafeGetUsername(credential);
+            string domain = NetworkCredentialHelper.UnsafeGetDomain(credential);
```

Comment:
This is all about partial trust code, which has gone away so regardless, I can remove the attribute. I suspect it is still technically classed as unsafe as it will do P/Invoke work on some platforms to get a security context for the currently logged in user but as we don't need to worry about partial trust, naming it unsafe doesn't gain us anything. I'll remove the attribute and rename a few things.